codegen/templates/code/controller/event_handlers.gotmpl

```diff
@@ -15,11 +15,12 @@ import (
 
 {{- range $resource := $.Resources }}
 
+// Handle events for the {{ $resource.Kind }} Resource
 type {{ $resource.Kind }}EventHandler interface {
-    Create(obj *{{ group_import_name $ }}.{{ $resource.Kind }}) error
-    Update(old, new *{{ group_import_name $ }}.{{ $resource.Kind }}) error
-    Delete(obj *{{ group_import_name $ }}.{{ $resource.Kind }}) error
-    Generic(obj *{{ group_import_name $ }}.{{ $resource.Kind }}) error
+    Create{{ $resource.Kind }}(obj *{{ group_import_name $ }}.{{ $resource.Kind }}) error
```

Comment:
At one point we discussed collapsing `Create` and `Update` into a single `Upsert` function, is this something we're still considering?

Reply:
this is now handled by the `Reconciler` interface, we should move over to that